File: src/main/java/com/EmosewaPixel/pixellib/materialSystem/materials/IMaterialItem.java
package com.EmosewaPixel.pixellib.materialsystem.materials;

import com.EmosewaPixel.pixellib.materialsystem.types.ObjectType;

//Tis interface is used for clases that take a material and object type
public interface IMaterialItem {
    Material getMaterial();

    ObjectType getObjType();
}

File: src/main/java/com/EmosewaPixel/pixellib/materialSystem/materials/IMaterialItem.kt
package com.EmosewaPixel.pixellib.materialsystem.materials

import com.EmosewaPixel.pixellib.materialsystem.types.ObjectType

//Tis interface is used for clases that take a material and object type
interface IMaterialItem {
    val mat: Material

    val objType: ObjectType
}